### PR TITLE
[MDB IGNORE] Removes tablet cartridges + reworks a ton more

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

- All tablets who previously had apps in a cartridge now has them built-into their tablet instead. This means it costs space for it.
- Rebalances the sizes of several apps to help them fit on Command tablets (Cargo ordering costed 20!!)
- Removes tablet cartridges, they've been reworked into a regular old portable disk (the same you use for toxins/borgs)
- Removes Signaller (the module required to run the signaller app) from tablets (likely will remove more in the future)
- Refactors the health/chem scanning app to not be as bad
- Dehardcodes detomatix resistance
- Ability to send PDA's to all is now tied to your access rather than a cartridge
- Moves 'eject disk' button to the very top of the UI

Old UI showing of how the disk works
https://cdn.discordapp.com/attachments/326831214667235328/968047635921502218/2022-04-25_03-14-37.mp4

New eject disk button at the top
https://cdn.discordapp.com/attachments/686626382083915787/968586208026439771/2022-04-26_14-55-14.mp4

Removes entirely:
- Detective cartridge
- Lawyer cartridge
- Janitor cartridge
- Curator cartridge
- Roboticist cartridge
- non-sensor Scientist cartridge

## Why It's Good For The Game

This is following the hackmd I wrote a while back, though not precisely as, for example, we're keeping the scanner stuff in https://hackmd.io/-523YJ5qRqigIS_zqns2pA?view

Cartridges aren't good for tablets because it bypasses one of the main restrictions they had: having to manage your storage space.
While I have given them more leeway in this PR by making apps cost less, we can tweak it more at a further date when tablets are in a better spot I feel (and hope).
The ability to steal applications has been kept in, as cartridges have been replaced with portable data disks that holds applications for all cartridges I thought actually needed them. Especially security, in which security officers do NOT have access to download the crew manifest app, something they SPAWN with.

Current Tablets were made to be a 1:1 copy of PDAs, basically just turning tablets INTO the PDAs, so instead of taking advantage of the tablet's new features to integrate PDA stuff into it, it just pasted it on top, gave it a new UI, and called it a day. There's many problems with tablets as a result of this and I hope that we can bring tablets back to what they could've been.

## Changelog

:cl:
del: The following job cartridges have been deleted: Detective, Lawyer, Janitor, Curator, Roboticist and Scientist (without sensor pack).
balance: The following tablet apps has had their storage space cost lowered Robocontrol 12 -> 6, Power monitor 9 -> 8, Budget ordering 20 -> 10, Power alarms 5 -> 4, Science hub 16 -> 10.
balance: You no longer need an MC Scanner part to use the tablet's remote signalling device application.
balance: Cartridges no longer let you run applications off of them, but you can still download applications off of them. Additionally, applications that were once on a roundstart cartridge (or were a program you had pre-Ntos PDAs) has been moved to be downloaded onto your tablet roundstart.
qol: The Eject Disk button on the tablet's UI has been moved to the very top to not clutter your UI as much.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
> Cartridges aren't good for tablets because it bypasses one of the main restrictions they had: having to manage your storage space.

it's not FUN

**Comment by JohnFulpWillard** [comment]
> > Cartridges aren't good for tablets because it bypasses one of the main restrictions they had: having to manage your storage space.
> 
> it's not FUN

Once again I'm gonna point to MODsuits, they have complexity that limits how many modules you can have at once. If you want a ton of storage space, ModPCs have it, at the cost of being immobile.
An Engineering Security Officer having all sec programs on top of all Engineering and Atmos tech programs isn't really "fun" for antagonists who will get chased by sec for causing a hull breach, triggering an alarm, or preventing sabotage because they're monitoring it 24/7 with this.

**Comment by MTandi** [comment]
Awesome, thanks! 
It has been bothering me since that merge, especially the value-pak of captain that had a dozen of programs that you could run on PDA even if they're not compatible with it (like power monitor).

**Comment by MTandi** [comment]
There's also an issue of space - the list of programs starts only from the middle of it, while the top part is occupied with less used elements. Maybe we should turn this disk panel into a simple button, similar to the "Eject ID" one.

image

**Comment by IndieanaJones** [comment]
> del: Detective, Lawyer, Janitor, Curator, Roboticist and Scientist (without sensor pack) has all been deleted.

You might want to re-word your changelog.  This probably doesn't sound like whatever you intended it to sound like.

**Comment by Dorsisdwarf** [comment]
> Once again I'm gonna point to MODsuits, they have complexity that limits how many modules you can have at once. 
OK and nobody enjoys having to mess with the modsuit settings, they _put up with that_ as the **cost** of getting to have a super cool-looking sidegrade to normal spacesuits. 

The weakest way to add fun to a game is to add "The fun of avoiding frustration" when the frustration is something that's just just made up out of thin air for the purpose of being avoided, IMO.

**Comment by san7890** [comment]
Is this the type of change we need an UpdatePaths script for, or is it just discarding a lot of stuff?

**Comment by JohnFulpWillard** [comment]
> Is this the type of change we need an UpdatePaths script for, or is it just discarding a lot of stuff? (this might also benefit from functionality introduced in #66506 if that gets in sooner rather than later)

yes, it updates paths.
it also deletes some so if that PR is merged before my own then I'll add it in.

**Comment by Thunder12345** [comment]
Tying PDA to all to access instead of a physical object feels like a bad idea to me. If a lawyer is abusing the feature to wind up the crew there's now no option to stop it other than confiscating their entire PDA or taking away their access, when currently you can just grab their PDA and pull the cartridge out to put a stop to it

**Comment by JohnFulpWillard** [comment]
> Tying PDA to all to access instead of a physical object feels like a bad idea to me. If a lawyer is abusing the feature to wind up the crew there's now no option to stop it other than confiscating their entire PDA or taking away their access, when currently you can just grab their PDA and pull the cartridge out to put a stop to it

You can possibly make it a configuration on the wirecarp app that the RD has, but I don't have any interest in having a physical item limit mass PDA messaging.

**Comment by Learner10** [comment]
If you're working on updating the tablets, can you confirm if the old skill checker ability of PDA's was ever moved over to tablets?

**Comment by JohnFulpWillard** [comment]
> If you're working on updating the tablets, can you confirm if the old skill checker ability of PDA's was ever moved over to tablets?

It's been  there for a while https://github.com/tgstation/tgstation/pull/63013

**Comment by Learner10** [comment]
> 

Ah, thanks. I couldn't find it last I looked.

File: tgui/packages/tgui/interfaces/NtosMain.js [hunk 1]
```diff
@@ -29,21 +29,39 @@ export const NtosMain = (props, context) => {
...
+                    <Button
+                      fluid
+                      icon="eject"
+                      content={device}
+                      onClick={() => act('PC_Eject_Disk', { name: device })}
+                      disabled={!device} />
+                  </Table.Cell>
+                ))}
+              </Table.Row>
+            </Table>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/NtosMain.js:64]: Please don't use tables for layout, we have `Stack` for this.
```js
<Stack>
  <Stack.Item>
    Flashlight
  </Stack.Item>
  <Stack.Item grow>
    Disk
  </Stack.Item>
</Stack>
```

File: tgui/packages/tgui/interfaces/NtosMain.js [hunk 2]
```diff
@@ -29,21 +29,39 @@ export const NtosMain = (props, context) => {
       width={400}
       height={500}>
       <NtosWindow.Content scrollable>
-        {!!has_light && (
+        {(!!has_light || !!removable_media.length) && (
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/NtosMain.js:32]: Would be more readable if you combined it like this:
```suggestion
        {Boolean(has_light || removable_media.length) && (
```

**Comment by private-tristan** [comment]
> Ability to send PDA's to all is now tied to your access rather than a cartridge
welp, that's sad, I always enjoyed getting a value cartridge and getting access to send all

**Comment by Mothblocks** [comment]
> Ability to send PDA's to all is now tied to your access rather than a cartridge

@LemonInTheDark is this really the world you want to live in

**Comment by JohnFulpWillard** [comment]
> > Tying PDA to all to access instead of a physical object feels like a bad idea to me. If a lawyer is abusing the feature to wind up the crew there's now no option to stop it other than confiscating their entire PDA or taking away their access, when currently you can just grab their PDA and pull the cartridge out to put a stop to it
> 
> You can possibly make it a configuration on the wirecarp app that the RD has, but I don't have any interest in having a physical item limit mass PDA messaging.

we live in a society where comments aren't listened to.

This is only meant to be part 1, I dont want to do everything in one PR.

**Comment by LemonInTheDark** [comment]
> > Ability to send PDA's to all is now tied to your access rather than a cartridge
> 
> @LemonInTheDark is this really the world you want to live in

Ehhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
no it's not.

I wonder if this should be handled like viruses,

**Comment by tralezab** [comment]
conflict (a lot, wow)

**Comment by LemonInTheDark** [comment]
I've thought about it some more.
Stuff that you want to be able to take away from someone should be tied to physical copies, or perhaps some way of changing what roles can download what programs. The first would do for now, but it is something we should handle before pushing this to live

**Comment by tralezab** [comment]
c-conflict again 😢

File: code/modules/modular_computers/file_system/programs/phys_scanner.dm [hunk 3]
```diff
@@ -1,3 +1,6 @@
+#define TABLET_MEDICAL_MODE (1<<0)
+#define TABLET_CHEMISTRY_MODE (1<<1)
```
Comment (SuperNovaa41) [inline code/modules/modular_computers/file_system/programs/phys_scanner.dm:2]: undef these

**Comment by SuperNovaa41** [comment]
conflicsts + review

**Pull request merged**